hera_cal/abscal.py

```diff
@@ -997,19 +1000,19 @@ def abscal_run(data_files, model_files, calfits_infiles=None, verbose=True, over
                 data = match_red_baselines(data, data_antpos, model, model_antpos, tol=tol, verbose=verbose)
                 antpos = model_antpos
 
-            # interpolate model to match data
-            if interp_model:
-                interp_model, interp_mflags = interp2d_vis(model, model_lsts, model_freqs,
-                                                           data_lsts, data_freqs, flags=model_flags)
-                for k in interp_mflags.keys():
-                    wgts[k][interp_mflags[k]] *= 0
+            # rephase model to match data lst grid
+            if rephase_model:
+                new_model, new_flags = rephase_vis(model, model_lsts, data_lsts, bls, data_freqs, flags=model_flags, latitude=latitude)
+
+                for k in new_flags.keys():
+                    wgts[k][new_flags[k]] *= 0
 
             # reweight according to redundancy
             if reweight:
                 wgts = mirror_data_to_red_bls(wgts, model_antpos, tol=tol, weights=True)
 
             # instantiate class
-            AC = AbsCal(interp_model, data, wgts=wgts, antpos=antpos, freqs=data_freqs)
+            AC = AbsCal(new_model, data, wgts=wgts, antpos=antpos, freqs=data_freqs)
```

Comment:
I think this will fail if rephase_model is False because then new_model will be undefined. Perhaps you need an `else: new_model = model` or something similar?

Reply:
good catch